delocate/tests/test_libsana.py

```diff
@@ -172,15 +185,31 @@ def test_wheel_libs():
                   realpath(LIBSYSTEMB): {mod2: LIBSYSTEMB}})
 
     def filt(fname):
+        # type: (Text) -> bool
         return not fname.endswith(mod2)
     assert_equal(wheel_libs(PLAT_WHEEL, filt), {})
 
 
 def test_resolve_rpath():
+    # type: () -> None
     # A minimal test of the resolve_rpath function
     path, lib = split(LIBA)
     lib_rpath = pjoin('@rpath', lib)
     # Should skip '/nonexist' path
-    assert_equal(resolve_rpath(lib_rpath, ['/nonexist', path]), realpath(LIBA))
-    # Should return the given parameter as is since it can't be found
-    assert_equal(resolve_rpath(lib_rpath, []), lib_rpath)
+    assert_equal(
+        resolve_rpath(lib_rpath, ['/nonexist', path], path), realpath(LIBA)
+    )
+    # Should raise DependencyNotFound if the dependency can not be resolved.
+    with pytest.raises(DependencyNotFound):
+        resolve_rpath(lib_rpath, [], path)
+
+
+def test_get_dependencies():
+    # type: () -> None
+    with pytest.raises(DependencyNotFound):
+        list(get_dependencies("nonexistent.lib"))
+    # This check might be too platform specific.
+    assert dict(get_dependencies(LIBA)) == {
+        realpath("/usr/lib/libstdc++.6.dylib"): "/usr/lib/libstdc++.6.dylib",
+        realpath("/usr/lib/libSystem.B.dylib"): "/usr/lib/libSystem.B.dylib",
+    }
```

Comment:
Consider direct tests for new functions:

* `walk_library`
* `walk_directory`